src/Services/Air/Air.js

```diff
@@ -127,72 +127,80 @@ module.exports = (settings) => {
-      return service.getUniversalRecordByPNR(options)
-        .catch((err) => {
-          // Checking for error type
-          if (!(err instanceof AirRuntimeError.NoReservationToImport)) {
-            return Promise.reject(err);
-          }
-          // Creating passive segment to import PNR
-          const terminal = createTerminalService(settings);
-          const segment = {
-            date: moment().add(42, 'days').format('DDMMM'),
-            airline: 'OK',
-            from: 'DOH',
-            to: 'ODM',
-            comment: 'NO1',
-            class: 'Y',
-          };
-          const segmentCommand = (
-            `0${segment.airline}OPEN${segment.class}${segment.date}${segment.from}${segment.to}${segment.comment}`
-          ).toUpperCase();
-          const segmentResult = (
-            `1. ${segment.airline} OPEN ${segment.class}  ${segment.date} ${segment.from}${segment.to} ${segment.comment}`
-          ).toUpperCase();
-          const pnrRegExp = new RegExp(`^${bookingInUseRegexp}${nonIataLineRegexp}${options.pnr}`);
-          return terminal.executeCommand(`*${options.pnr}`)
-            .then((response) => {
-              if (!response.match(pnrRegExp)) {
-                return Promise.reject(new AirRuntimeError.UnableToOpenPNRInTerminal());
-              }
-              return Promise.resolve();
-            })
-            .then(() => terminal.executeCommand(segmentCommand))
-            .then((response) => {
-              if (response.indexOf(segmentResult) === -1) {
-                return Promise.reject(new AirRuntimeError.UnableToAddExtraSegment());
-              }
-              return Promise.resolve();
-            })
-            .then(() => {
-              const ticketingDate = moment().add(10, 'days').format('DDMMM');
-              const command = `T.TAU/${ticketingDate}`;
-              return terminal.executeCommand(command)
-                .then(() => terminal.executeCommand('R.UAPI+ER'))
-                .then(() => terminal.executeCommand('ER'));
-            })
-            .then((response) => {
-              if (
-                (!response.match(pnrRegExp))
-                || (response.indexOf(segmentResult) === -1)
-              ) {
-                return Promise.reject(new AirRuntimeError.UnableToSaveBookingWithExtraSegment());
-              }
-              return Promise.resolve();
-            })
-            .catch(
-              importErr => terminal.closeSession().then(
-                () => Promise.reject(
-                  new AirRuntimeError.UnableToImportPnr(options, importErr)
-                )
-              )
-            )
-            .then(() => terminal.closeSession())
-            .then(() => service.getUniversalRecordByPNR(options))
-            .then(ur => service.cancelBooking(getBookingFromUr(ur, options.pnr)))
-            .then(() => service.getUniversalRecordByPNR(options));
-        });
+    async handleNoReservationError(options) {
```

Comment:
Not sure about the function name as we get a valid booking in case of success result and typed an error in any other case. What about splitting this function to createBookingPassiveSegment (passive segment creation block), and this block 
```
      await terminal.closeSession();
      const ur = await service.getUniversalRecordByPNR(options);
      await service.cancelBooking(getBookingFromUr(ur, options.pnr));

      return service.getUniversalRecordByPNR(options);
```
Let's discuss it later tomorrow.